### PR TITLE
[DCOS_OSS-1705] Make SSH run and popen user configurable

### PR DESCRIPTION
**Pull request by timaa2k**
https://jira.mesosphere.com/browse/DCOS_OSS-1705

### CODE REVIEW

File: tests/test_node.py [hunk 1]
```diff
@@ -90,18 +177,49 @@ def test_popen_as_root(
...
+            assert cp_result.returncode == 0
+            chown_result = master.run(
+                args=['chown', '-R', 'testuser', '/home/testuser/.ssh'],
+                user='root'
+            )
+            assert chown_result.returncode == 0
+            # Remove stray file that prevents non-root SSH.
+            # https://ubuntuforums.org/showthread.php?t=2327330
+            rm_result = master.run(
+                args=['rm', '-f', '/run/nologin'], user='root'
```
Comment (adamtheturtle) [inline tests/test_node.py]: @timaa2k Do you agree that this should be done automatically? Maybe at the `cluster` level, or maybe at the DC/OS Docker backend level.

Comment (timaa2k) [reply tests/test_node.py]: I think it should be done at the specific backend level. dcos-e2e should require this for every potential backend in order to guarantee that it provides the run as user functionality.

**Comment by timaa2k** [comment]
Please review @adamtheturtle

File: API.md [hunk 2]
```diff
@@ -243,3 +257,4 @@ The method returns a `Popen` object that can be used to communicate to the under
 #### `ip_address`
 
 The IP address of the node.
+
```
Comment (adamtheturtle) [inline API.md]: Remove excess line.

File: README.md [hunk 3]
```diff
@@ -59,11 +59,11 @@ class TestExample:
             generate_config_path=Path('/tmp/dcos_generate_config.sh'),
         ) as cluster:
             (master, ) = cluster.masters
-            result = master.run_as_root(args=['test', '-f', path])
+            result = master.run(args=['test', '-f', path], user='root')
```
Comment (adamtheturtle) [inline README.md]: Revert changes to this file.

File: src/dcos_e2e/cluster.py [hunk 4]
```diff
@@ -120,8 +120,9 @@ def wait_for_dcos(self) -> None:
         ]
 
         for node in self.masters:
-            node.run_as_root(
+            node.run(
```
Comment (adamtheturtle) [inline src/dcos_e2e/cluster.py]: Revert changes to this file.

File: src/dcos_e2e/node.py [hunk 5]
```diff
@@ -31,13 +31,15 @@ def __init__(self, ip_address: IPv4Address, ssh_key_path: Path) -> None:
...
         args: List[str],
+        user: str,
         env: Optional[Dict]=None,
     ) -> List[str]:
         """
         Run the specified command on the given host using SSH.
 
         Args:
             args: The command to run on the node.
+            user: The username to SSH.
```
Comment (adamtheturtle) [inline src/dcos_e2e/node.py]: This is ambiguous - it maybe says that `user` will perform the SSH command. Make it clear that this is the username for a user on the node which will run the command.

File: tests/test_harness.py [hunk 6]
```diff
@@ -106,7 +106,7 @@ def test_extend_config(
         ) as cluster:
             cluster.wait_for_dcos()
             (master, ) = cluster.masters
-            master.run_as_root(args=['test', '-f', path])
+            master.run(args=['test', '-f', path], user='root')
```
Comment (adamtheturtle) [inline tests/test_harness.py]: Revert the changes which are unnecessary.

File: src/dcos_e2e/node.py [hunk 7]
```diff
@@ -31,13 +31,15 @@ def __init__(self, ip_address: IPv4Address, ssh_key_path: Path) -> None:
...
         args: List[str],
+        user: str,
         env: Optional[Dict]=None,
     ) -> List[str]:
         """
         Run the specified command on the given host using SSH.
 
         Args:
             args: The command to run on the node.
+            user: The the user on the given host that the command will be run for over SSH.
```
Comment (adamtheturtle) [inline src/dcos_e2e/node.py]: `s/The the/The`.

No need to wait for tests for this change.

**Comment by adamtheturtle** [comment]
Thanks @timaa2k and @gpaul for a really useful change!
